src/elements/dropdown/dropdown.js

```
@@ -0,0 +1,455 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('semantic.ui.elements.dropdown', [])
+    .directive('smDropdown', smDropdown)
+    .directive('smDropdownItem', smDropdownItem)
+    .directive('smDropdownSearch', smDropdownSearch);
+
+  var keys = {
+    enter      : 13,
+    space      : 32,
+    escape     : 27,
+    leftArrow  : 37,
+    upArrow    : 38,
+    rightArrow : 39,
+    downArrow  : 40
+  };
+
+  function smDropdown() {
+    return {
+      restrict: 'E',
+      require: ['smDropdown', '?ngModel'],
+      compile: compile,
+      controller: DropdownCtrl
+    };
+
+    function compile(element, attrs) {
+      var
+        labelElm = element.find('sm-dropdown-label'),
+        menuElm = element.find('sm-dropdown-menu');
+
+      labelElm.addClass('text');
+
+      menuElm.addClass('menu transition');
+      menuElm.attr('tabindex', -1);
+
+      element.addClass('ui dropdown');
+      if (angular.isDefined(attrs.type)) {
+        element.addClass(attrs.type);
+      }
+
+      if (isType('selection')) {
+        labelElm.addClass('default');
+      }
+
+      if (isType('search')) {
+        element.append('<input class="search" tabindex="0" sm-dropdown-search>');
+      } else {
+        element.attr('tabindex', 0);
+      }
+
+      return postLink;
+
+      function isType(type) {
+        return attrs.type && attrs.type.indexOf(type) > -1;
+      }
+    }
+
+    function postLink(scope, element, attrs, ctrls) {
+      var
+        dropdownCtrl = ctrls[0],
+        ngModel = ctrls[1];
```

Should we throw/warn when ng-model was not provided ?